frontend/public/components/cluster-service-class-icon.tsx

```diff
@@ -6,13 +6,109 @@ import * as classNames from 'classnames';
+
+const LOGOS = {
+  'icon-3scale': '3scale.svg',
+  'icon-aerogear': 'aerogear.svg',
+  'icon-amq': 'amq.svg',
+  'icon-angularjs': 'angularjs.svg',
+  'icon-ansible': 'ansible.svg',
+  'icon-apache': 'apache.svg',
+  'icon-beaker': 'beaker.svg',
+  'icon-capedwarf': 'capedwarf.svg',
+  'icon-cassandra': 'cassandra.svg',
+  'icon-clojure': 'clojure.svg',
+  'icon-codeigniter': 'codeigniter.svg',
+  'icon-cordova': 'cordova.png',
+  'icon-datagrid': 'datagrid.svg',
+  'icon-datavirt': 'datavirt.svg',
+  'icon-debian': 'debian.svg',
+  'icon-decisionserver': 'decisionserver.svg',
+  'icon-django': 'django.svg',
+  'icon-dotnet': 'dotnet.svg',
+  'icon-drupal': 'drupal.svg',
+  'icon-eap': 'eap.svg',
+  'icon-elastic': 'elastic.svg',
+  'icon-erlang': 'erlang.svg',
+  'icon-fedora': 'fedora.svg',
+  'icon-freebsd': 'freebsd.svg',
+  'icon-git': 'git.svg',
+  'icon-github': 'github.svg',
+  'icon-gitlab': 'gitlab.svg',
+  'icon-glassfish': 'glassfish.svg',
+  'icon-go-gopher': 'go-gopher.svg',
+  'icon-grails': 'grails.svg',
+  'icon-hadoop': 'hadoop.svg',
+  'icon-haproxy': 'haproxy.svg',
+  'icon-infinispan': 'infinispan.svg',
+  'icon-jboss': 'jboss.svg',
+  'icon-jenkins': 'jenkins.svg',
+  'icon-jetty': 'jetty.svg',
+  'icon-joomla': 'joomla.svg',
+  'icon-jruby': 'jruby.svg',
+  'icon-js': 'js.svg',
+  'icon-kubevirt': 'kubevirt.svg',
+  'icon-laravel': 'laravel.svg',
+  'icon-load-balancer': 'load-balancer.svg',
+  'icon-mariadb': 'mariadb.svg',
+  'icon-mediawiki': 'mediawiki.svg',
+  'icon-memcached': 'memcached.svg',
+  'icon-mongodb': 'mongodb.svg',
+  'icon-mssql': 'mssql.svg',
+  'icon-mysql-database': 'mysql-database.svg',
+  'icon-nginx': 'nginx.svg',
+  'icon-nodejs': 'nodejs.svg',
+  'icon-openjdk': 'openjdk.svg',
+  'icon-openshift': 'openshift.svg',
+  'icon-openstack': 'openstack.svg',
+  'icon-other-linux': 'other-linux.svg',
+  'icon-other-unknown': 'other-unknown.svg',
+  'icon-perl': 'perl.svg',
+  'icon-phalcon': 'phalcon.svg',
+  'icon-php': 'php.svg',
+  'icon-play': 'play.svg',
+  'icon-postgresql': 'postgresql.svg',
+  'icon-processserver': 'processserver.svg',
+  'icon-python': 'python.svg',
+  'icon-rabbitmq': 'rabbitmq.svg',
+  'icon-rails': 'rails.svg',
+  'icon-redis': 'redis.svg',
+  'icon-rh-integration': 'rh-integration.svg',
+  // Use the upstream icon.
+  'icon-rh-openjdk': 'openjdk.svg',
+  'icon-rh-tomcat': 'rh-tomcat.svg',
+  'icon-ruby': 'ruby.svg',
+  'icon-scala': 'scala.svg',
+  'icon-shadowman': 'shadowman.svg',
+  'icon-spring': 'spring.svg',
+  'icon-sso': 'sso.svg',
+  'icon-stackoverflow': 'stackoverflow.svg',
+  'icon-suse': 'suse.svg',
+  'icon-symfony': 'symfony.svg',
+  'icon-tomcat': 'tomcat.svg',
+  'icon-ubuntu': 'ubuntu.svg',
+  'icon-wildfly': 'wildfly.svg',
+  'icon-windows': 'windows.svg',
+  'icon-wordpress': 'wordpress.svg',
+  'icon-xamarin': 'xamarin.svg',
+  'icon-zend': 'zend.svg'
+};
+
+const imageForIconClass = iconClass => {
+  return _.get(LOGOS, iconClass);
```

Comment:
You would want to use `_.get(LOGOS, [iconClass])` in case iconClass had special characters. But _.get isn't needed here because we know LOGOS is defined

```js
return LOGOS[iconClass];
```

I'd probably just inline that honestly.